### PR TITLE
ci: storage: publish SPDK container to ghcr

### PR DESCRIPTION
**Pull request by glimchb**
- move SPDK docker to storage
- ci: storage: publish SPDK container to ghcr

### CODE REVIEW

**Review by mestery — CHANGES_REQUESTED** [review]
This looks ok, but don't we want to publish all the Dockerfiles in this repo?

File: .github/workflows/poc-storage.yml [hunk 1]
```diff
@@ -13,6 +13,12 @@ on:
       - 'storage/**'
       - '.github/workflows/poc-storage.yml'
 
+env:
+  # Use docker.io for Docker Hub if empty
+  REGISTRY: ghcr.io
+  # github.repository as <account>/<repo>
+  IMAGE_NAME: ${{ github.repository }}
```
[the remark this answers is not in the document]
Comment (glimchb) [reply .github/workflows/poc-storage.yml:20]: yep, was thinking `opi-storage` or `opi-spdk` maybe... WDYT ?

[the remark this answers is not in the document]
Comment (mestery) [reply .github/workflows/poc-storage.yml:20]: I like that!

[the remark this answers is not in the document]
Comment (mestery) [reply .github/workflows/poc-storage.yml:20]: Lets make the change before we merge so as you add other ones in they don't cause issues with GHCR storage and such.

[the remark this answers is not in the document]
Comment (glimchb) [reply .github/workflows/poc-storage.yml:20]: on it

File: .github/workflows/poc-storage.yml [hunk 2]
```diff
@@ -13,6 +13,12 @@ on:
       - 'storage/**'
       - '.github/workflows/poc-storage.yml'
 
+env:
+  # Use docker.io for Docker Hub if empty
+  REGISTRY: ghcr.io
+  # github.repository as <account>/<repo>
+  IMAGE_NAME: ${{ github.repository_owner }}/opi-spdk
```
Comment (mestery) [inline .github/workflows/poc-storage.yml:20]: Love it!

File: storage/docker-compose.yml [hunk 3]
```diff
@@ -3,6 +3,7 @@ version: "3.7"
 services:
 
   spdk:
+    image: ghcr.io/opiproject/opi-spdk:main
```
Comment (mestery) [inline storage/docker-compose.yml:6]: ❤️

**Review by mestery — APPROVED** [review]
Amazing work! Let's go!

**Pull request merged**